tests/smartclone_test.go

```diff
@@ -0,0 +1,97 @@
+package tests
+
+import (
+	"fmt"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/onsi/ginkgo/extensions/table"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"kubevirt.io/containerized-data-importer/pkg/controller"
+	"kubevirt.io/containerized-data-importer/tests/framework"
+	"kubevirt.io/containerized-data-importer/tests/utils"
+
+	cdiv1 "kubevirt.io/containerized-data-importer/pkg/apis/core/v1alpha1"
+)
+
+var _ = Describe("[vendor:cnv-qe@redhat.com][level:component]SmartClone tests", func() {
+
+	var sourcePvc *v1.PersistentVolumeClaim
+
+	fillData := "123456789012345678901234567890123456789012345678901234567890"
+	testFile := utils.DefaultPvcMountPath + "/source.txt"
+	fillCommand := "echo \"" + fillData + "\" >> " + testFile
+	rbdStorageClassName := "csi-rbd"
+
+	f := framework.NewFrameworkOrDie("dv-func-test")
+
+	AfterEach(func() {
+		if sourcePvc != nil {
+			By("[AfterEach] Clean up target PVC")
+			err := f.DeletePVC(sourcePvc)
+			Expect(err).ToNot(HaveOccurred())
+			sourcePvc = nil
+		}
+	})
+
+	Describe("Verify DataVolume Smart Cloning", func() {
+		table.DescribeTable("should", func(name, command, url, dataVolumeName, eventReason string, phase cdiv1.DataVolumePhase) {
+			if controller.IsOpenshift(f.K8sClient) {
+				Skip("Test not available in openshift")
+			}
+			var dataVolume *cdiv1.DataVolume
+			switch name {
+			case "smart-clone":
+				sourcePVCName := fmt.Sprintf("%s-src-pvc", dataVolumeName)
+				sourcePodFillerName := fmt.Sprintf("%s-filler-pod", dataVolumeName)
+				sourcePvc = f.CreateAndPopulateSourcePVC(sourcePVCName, sourcePodFillerName, command, "csi-rbd")
+
+				By(fmt.Sprintf("creating a new target PVC (datavolume) to clone %s", sourcePvc.Name))
+				dataVolume = utils.NewCloningDataVolume(dataVolumeName, "1Gi", sourcePvc)
+				dataVolume.Spec.PVC.StorageClassName = &rbdStorageClassName
+			}
+
+			By(fmt.Sprintf("creating new datavolume %s", dataVolume.Name))
+			dataVolume, err := utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, dataVolume)
+			Expect(err).ToNot(HaveOccurred())
+
+			By(fmt.Sprintf("waiting for datavolume to match phase %s", string(phase)))
+			utils.WaitForDataVolumePhase(f.CdiClient, f.Namespace.Name, phase, dataVolume.Name)
```

Comment:
You didn't assign err, so the following check is against the err above.

Reply:
Removed the Expect.
We have the same issue here:
https://github.com/kubevirt/containerized-data-importer/blob/master/tests/datavolume_test.go#L91